src/cdk/testing/protractor/protractor-harness-environment.ts

```diff
@@ -24,8 +24,9 @@ export class ProtractorHarnessEnvironment extends HarnessEnvironment<ElementFind
   async forceStabilize(): Promise<void> {}
 
   async waitForTasksOutsideAngular(): Promise<void> {
-    // TODO: figure out how we can do this for the protractor environment.
-    // https://github.com/angular/components/issues/17412
+    await browser.executeAsyncScript(
+      'var callback = arguments[arguments.length - 1];'
+      + 'requestAnimationFrame(callback);');
```

Comment:
I think we need to spend more time thinking about this. e.g. `requestAnimationFrame` does not wait for macro tasks (as our task interceptor does in the testbed environment).

Reply:
It doesn't? My first attempt used setTimout, but I thought rAF waited for both micro and macro task queues to clear.

Reply:
If so, we could call the async script callback only after setTimeout(..., 1) and requestAnimationFrame has cleared.

Reply:
I think we should be able to use a similar strategy to what we did for the TestbedHarnessEnvironment, we'll just need to run the task interceptor as a browser script

Reply:
@LayZeeDK I might miss something here, but won't things like `setTimeout` and `requestAnimationFrame` never work here because we don't know what durations the macro tasks take? i.e. the component could have something like `setTimeout(..., 300)` inside. 

This is why we need ZoneJS here since it can capture that information for us. I think what @mmalerba proposed seems right. I'm unsure how it will work yet, but it should be possible.

Reply:
Alright, so you want to patch the environment with zones or similar like the TestBed harness environment.

Two issues which might be an entirely different discussion of their own:
1. With ES2017+ bundles, it's not possible to intercept async-await.
2. Can we support applications using NgNoopZone with the task interceptor?

Reply:
I think for this PR we should just focus on getting protractor working with es2015 and using `NgZone`. That's what we currently support for the `TestbedHarnessEnvironment`.

1. Yeah, that is a separate issue that affects many parts of Angular, it is being tracked here: https://github.com/angular/angular/issues/31730
2. Currently we have not tested it with `NoopNgZone` and my guess is it won't work. If you have thoughts/ideas about this please feel free to open a new issue for the feature and discuss them there.

Reply:
Is v9 going go be released without this implemented?